src/ck3/data/courtier_guest_mgmt.rs

```diff
@@ -0,0 +1,88 @@
+        let mut vd = Validator::new(block, data);
+
+        if key.is("courtier_management") {
+            let sc_builder = |key: &Token| {
+                let mut sc = ScopeContext::new(Scopes::Character, key);
+                sc.define_name("courtier", Scopes::Character, key);
+                sc.define_name("liege", Scopes::Character, key);
+                sc
+            };
+
+            vd.field_validated_key_block("can_leave", |key, block, data| {
+                validate_trigger(block, data, &mut sc_builder(key), Tooltipped::No);
+            });
+            vd.field_validated_key_block("monthly_leave_chance_x10", |key, block, data| {
+                validate_modifiers_with_base(block, data, &mut sc_builder(key));
+            });
+        } else if key.is("guest_management") {
+            let sc_builder = |key: &Token| {
+                let mut sc = ScopeContext::new(Scopes::Character, key);
+                sc.define_name("guest", Scopes::Character, key);
+                sc.define_name("host", Scopes::Character, key);
+                sc
+            };
+            vd.field_validated_key_block("guest_can_arrive", |key, block, data| {
+                validate_trigger(block, data, &mut sc_builder(key), Tooltipped::No);
+            });
+            vd.field_validated_key_block("guest_score", |key, block, data| {
+                validate_modifiers_with_base(block, data, &mut sc_builder(key));
+            });
+            vd.field_validated_key_block("can_leave", |key, block, data| {
+                validate_trigger(block, data, &mut sc_builder(key), Tooltipped::No);
+            });
+            vd.field_validated_key_block("monthly_leave_chance_x10", |key, block, data| {
+                validate_modifiers_with_base(block, data, &mut sc_builder(key));
+            });
+            vd.field_validated_block("guest_description", |block, data| {
+                let mut vd = Validator::new(block, data);
+                vd.multi_field_validated_block("description", |block, data| {
+                    let mut vd = Validator::new(block, data);
+                    vd.req_field("desc");
+                    vd.field_validated_key_block("limit", |key, block, data| {
+                        validate_trigger(block, data, &mut sc_builder(key), Tooltipped::No);
+                    });
+                    vd.field_validated_key_block("weight", |key, block, data| {
+                        validate_modifiers_with_base(block, data, &mut sc_builder(key));
+                    });
+                    vd.field_validated_key("desc", |key, bv, data| {
+                        validate_desc(bv, data, &mut sc_builder(key));
+                    });
+                });
+            });
+        } else {
+            let msg = "expected either `courtier_management` or `guest_management`";
+            err(ErrorKey::Validation).msg(msg).loc(key).push();
+            return;
```

Comment:
don't see why we need the `return` there

Reply:
Ah yeah that was left over after restructuring the code